### DISCUSSION TITLE
Advanced theming not working properly with NextJS

### DISCUSSION THREAD

**Discussion by ptrhck** [ben-rogerson/twin.macro#299] [Q&A]
I have setup the [next-styled-components](https://github.com/ben-rogerson/twin.examples/tree/master/next-styled-components) example and wanted to tweak it to work with [advanced-theming](https://github.com/ben-rogerson/twin.macro/blob/master/docs/advanced-theming.md). I had a look at the react + styled-components example. Instead of the custom hook, I have used the [next-themes](https://github.com/pacocoursey/next-themes) hook. However, the CSS is not applied correctly. What am I missing?

See Codesandbox [here](https://codesandbox.io/s/kkebn?file=/pages/_app.tsx) :)

**Comment by ptrhck** [comment]
The following works though:
```
import { ThemeToggle, Logos, StylesBase } from '../components'
import { GlobalStyles } from 'twin.macro'
const App = () => (
  <div>
    <GlobalStyles/>
    <StylesBase />
    <div tw="h-screen flex flex-col items-center">
      <Logos />
      <div tw="h-full flex justify-center items-center">
        <ThemeToggle />
      </div>
      <a
        tw="mb-10 hover:underline"
        href="https://alexperronnet.io"
        target="_blank"
        rel="noopener noreferrer"
      >
        @alexperronnet
      </a>
    </div>
  </div>
)

export default App
```
What is the GlobalStyles macro for?

**Reply by ben-rogerson** [reply]
Great you got it working 👍 

`<GlobalStyles />` adds the base styles required for tailwind classes to function correctly, you'll need that.

**Reply by ptrhck** [reply]
Thank you!

Yes it is working, but should `<GlobalStyles />` not be in [_app.js](https://github.com/ben-rogerson/twin.examples/blob/master/next-styled-components/pages/_app.js) so it is applied to all sites?

**Reply by ben-rogerson** [reply]
Sorry, not sure what you mean. It's added to _app.js so it's only loaded once in your project.

**Reply by ptrhck** [reply]
Sorry for my missleading answer. The codebox example [above](https://codesandbox.io/s/kkebn?file=/pages/_app.tsx), where it is added to the _app.tsx file, does not display it correctly. If I move it to the index file, as shown in my second post, it is displayed correctly (see sandbox [here](https://codesandbox.io/s/long-glade-dng63?file=/pages/index.tsx)).

**Reply by ben-rogerson** [reply]
Sounds like [it could be this issue](https://github.com/ben-rogerson/twin.macro/issues/277) you're seeing, the fix is to move the import to another file and import that new file into one of your root project components.

**Reply by ptrhck** [reply]
Well, that is what I did as shown in your next-styled-components example. As you can see [here](https://codesandbox.io/s/kkebn?file=/components/GlobalStyles.tsx), it is already in another file.
